Sources/MapboxMaps/Camera/BasicCameraAnimatorImpl.swift

```diff
@@ -171,7 +171,11 @@ internal final class BasicCameraAnimatorImpl: BasicCameraAnimatorProtocol {
     internal func stopAnimation() {
         switch internalState {
         case .initial:
-            fatalError("Attempt to stop an animation that has not started.")
+            internalState = .final
+            for completion in completions {
+                completion(.current)
+            }
+            completions.removeAll()
```

Comment:
I'll review this behavior again to see if we depend on it in any way and whether we've been consistent across releases.

Reply:
is it also intended that this will not set the state of `propertyAnimator` to `inactive` or `stopped`? Since we have now our own internal state of the animator, is there a guarantee that this internal state will be in sync with the properyAnimator' state?

Reply:
In all Maps SDK 10.x versions since the final v10.0.0 release, we've had the following behavior:

* BasicCameraAnimator - cancel allowed unless not started; completions invoked with .current (unless already finished)
* FlyToCameraAnimator - cancel allowed at any time; completions invoked with .current (unless already finished)
* GestureDecelerationCameraAnimator - cancel allowed at any time; completion invoked (no argument) (unless already finished)

Probably the biggest reason for this design is that we don't distinguish between SDK-initiated cancellation (e.g. canceled due to gesture input) and developer-initiated cancellation. `DefaultViewportTransition` is the only component I see in the SDK that depends on completion blocks being invoked if animations are canceled.

I think we should keep the behavior as-is for now, but we could reevaluate it for v11. Maybe we would want to invoke completion blocks if the animations are interrupted for reasons like gesture input but not if they were canceled explicitly by the developer. It's a tricky situation though because `DefaultViewportTransition` would probably still need to know if its animations were interrupted by the developer, and I could imagine creating other components like that in the future.

@maios I'll reply to your question in a separate comment

Reply:
@macdrevx Thanks for the thorough review, indeed, it makes sense to keep it as-is.

Reply:
@maios the code highlighted here only runs if the animator's internal state is `.initial` in which case the property animator hasn't been started yet and is still in the `.inactive` state. The current API expects animators `state` to be `.inactive` before they are started and after they are canceled or completed, so no change to the property animator state is required in this situation.

As far as a guarantee about keeping internalState and propertyAnimator.state in sync — there are none other than that we have to be vigilant to ensure the state machine keeps they in sync correctly. It's one of many major downsides to this implementation.

Looking forward to v11, I would love to replace the UIViewPropertyAnimator/CameraView implementation with a custom interpolation solution. A first step in that direction is in the works as part of the solution to what we've been calling the "moving target problem" of the viewport API. One of my next PRs will introduce a new internal-only animator that has a subset of the functionality offered by `BasicCameraAnimator` implemented without piggybacking on Core Animation. Introducing and using this in a limited capacity will give us the opportunity to start proving out this approach and if all goes well, we can grow its capabilities to be able to replace BasicCameraAnimator's implementation in v11.